### DISCUSSION TITLE
Caddy proxy: Route unknown hosts to first app?

### DISCUSSION THREAD

**Discussion by timsluis** [dokku/dokku#5541] [Q&A]
By default Dokku will route any received request with an unknown HOST header value to the lexicographically first site in the Nginx config stack.

With Caddy, unknown hosts will get a `ERR_SSL_PROTOCOL_ERROR`.

Can we recreate the Nginx behaviour for Caddy as well?

A little more context: a very nice feature of Caddy is on-demand TLS. With just a little extra configuration, Caddy can make it very easy to create multi-tenant apps in Dokku if we could route unknown hosts to a single app 😎.

**Comment by josegonzalez** [comment]
The functionality that we exhibit with the nginx plugin is due to how nginx works (which personally I think is a terrible default).

I don't know how to produce the same functionality with Caddy, but I suspect you'll have better luck asking in a Caddy forum as to how to configure a catch-all host. Once you have that, it should be possible for us to transform that into the requisite labels for usage with our Caddy integration.

**Reply by timsluis** [reply]
Thanks! Will look into it soon.
